### PR TITLE
Deprecate domain card

### PR DESCRIPTION
**Pull request by cursedcoder**
## Hey, I just made a Pull Request!

See issue #21742.

Tl;dr this deprecates the duplicated code.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by tudi2d — APPROVED** [review]
Approving as I do agree & think this is a good change, but I remember that there was a reason that this exists. Maybe someone of the maintainers can shed a light on this.

File: plugins/explore/src/components/DomainCard/DomainCard.tsx [hunk 1]
```diff
@@ -34,7 +34,10 @@ import React from 'react';
 import { LinkButton, ItemCardHeader } from '@backstage/core-components';
 import { useRouteRef } from '@backstage/core-plugin-api';
 
-/** @public */
+/**
+ * @public
+ * @deprecated You should use EntityCard instead, it's identical
```
Comment (awanlin) [inline plugins/explore/src/components/DomainCard/DomainCard.tsx:39]:
```suggestion
 * @deprecated You should use {@link EntityCard} instead, it's identical
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @cursedcoder, just left one comment.

For history sake the `DomainCard` came first and then the generic `EntityCard` was added as part of this PR: https://github.com/backstage/backstage/pull/15381 @tudi2d

File: .changeset/large-stingrays-try.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-explore': minor
```
Comment (benjdlambert) [inline .changeset/large-stingrays-try.md:2]: Think this should be a `patch` release as the package is 0.x and this isn't a breaking change.

**Review by benjdlambert — DISMISSED** [review]
Nice! Thanks for this! :tada:

**Comment by benjdlambert** [comment]
@cursedcoder sorry, I think that the API report is failing - I think it's because the `EntityCard` isn't exported.  Do we think that `EntityCard` is the right name for this? Or is there something better /cc @awanlin @tudi2d

**Comment by benjdlambert** [comment]
Looks like you'll need to run prettier on the repo to get [CI / Verify](https://github.com/backstage/backstage/actions/runs/7489202423/job/20385309881?pr=22060) to pass.

Also thinking if it's worth deleting the `DomainCard` implementation entirely, and just exporting the `EntityCard` as `DomainCard` but deprecate that export if it is exactly the same. There's no point in having duplicates of the same code right?

**Comment by cursedcoder** [comment]
@benjdlambert I don't like that idea.

About prettier - it's not prettier but some bug in the master branch. See other PRs also have it e.g. https://github.com/backstage/backstage/pull/22224

```
node_modules/@mui/base/Unstable_Popup/Popup.d.ts:13:70 - error TS2707: Generic type 'ElementType' requires between 0 and 1 type arguments.

13 declare const Popup: React.ForwardRefExoticComponent<Omit<PopupProps<React.ElementType<any, keyof React.JSX.IntrinsicElements>>, "ref"> & React.RefAttributes<Element>>;
                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@mui/material/Popper/Popper.d.ts:46:17 - error TS2707: Generic type 'ElementType' requires between 0 and 1 type arguments.

46     component?: React.ElementType<any, keyof React.JSX.IntrinsicElements> | undefined;
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@mui/material/Popper/Popper.d.ts:53:16 - error TS2707: Generic type 'ElementType' requires between 0 and 1 type arguments.

53         Root?: React.ElementType<any, keyof React.JSX.IntrinsicElements> | undefined;
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by benjdlambert** [comment]
@cursedcoder what I mean is to delete the `DomainCard` implementation and do this instead:

```ts
/** @deprecated **/
export const DomainCard = EntityCard
```

That way the export is still there, and deprecated, and we've deleted duplicate code if they're the same?

Can you elaborate why you don't like this idea?

**Comment by benjdlambert** [comment]
It seems like you fixed prettier anyways so that's fine. The errors you're referring to is the E2E test, which are broken because of this: https://github.com/mui/material-ui/issues/40427

**Pull request closed without merging**